src/goto-programs/show_goto_functions.cpp

```diff
@@ -41,36 +44,15 @@ void show_goto_functions(
   {
   case ui_message_handlert::XML_UI:
   {
-    //This only prints the list of functions
-    xmlt xml_functions("functions");
-    forall_goto_functions(it, goto_functions)
-    {
-      xmlt &xml_function=xml_functions.new_element("function");
-      xml_function.set_attribute("name", id2string(it->first));
-    }
-
-    std::cout << xml_functions << std::endl;
+    show_goto_functions_xmlt xml_show_functions(ns);
+    xml_show_functions.print_goto_functions(goto_functions, std::cout);
```

Comment:
Rationale: There are two public methods on this class, one prints out the XML functions to a provided `ostream` and one produces an XML object that the caller can do with what they will. 

These could both be `operator()` functions since they would have unique signatures. I thought the two names made the intention more explicit, but happy to change if you disagree.